images/gpu-feature-discovery/tests/main.tf

```diff
@@ -0,0 +1,83 @@
+terraform {
+  required_providers {
+    oci       = { source = "chainguard-dev/oci" }
+    imagetest = { source = "chainguard-dev/imagetest" }
+  }
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+variable "name" {
+  default = "gpu-feature-discovery"
+}
+
+data "oci_string" "ref" { input = var.digest }
+
+data "imagetest_inventory" "this" {}
+
+resource "imagetest_harness_k3s" "this" {
+  name      = var.name
+  inventory = data.imagetest_inventory.this
+
+  sandbox = {
+    mounts = [
+      {
+        source      = path.module
+        destination = "/tests"
+      }
+    ]
+  }
+}
+
+module "helm" {
+  source    = "../../../tflib/imagetest/helm"
+  name      = "nvgfd"
+  repo      = "https://nvidia.github.io/gpu-feature-discovery"
+  chart     = "gpu-feature-discovery"
+  namespace = "gpu-feature-discovery"
+
+  values = {
+    image = {
+      repository = data.oci_string.ref.registry_repo
+      tag        = "unused@${element(split("@", var.digest), 1)}"
+    }
+  }
+}
+
+resource "imagetest_feature" "basic" {
+  harness     = imagetest_harness_k3s.this
+  name        = "Basic"
+  description = "Basic functionality of the image."
+
+  steps = [
+    {
+      name = "Helm install"
+      cmd  = module.helm.install_cmd
+    },
+    {
+      name  = "Helm install GPU Feature Discovery"
+      cmd   = <<EOF
+        helm repo add nvgfd https://nvidia.github.io/gpu-feature-discovery
+        helm repo update
+        helm upgrade -i nvgfd nvgfd/gpu-feature-discovery \
+          --version 0.8.2 \
+          --namespace gpu-feature-discovery \
+          --create-namespace
+      EOF
+      retry = { attempts = 3, delay = "10s" }
+    },
```

Comment:
Isn't this captured by the `module.helm.install_cmd` step before this one?

Reply:
I thought so too. But doesn't work like that for some reason

Reply:
I removed this block and the tests worked locally. Keeping the cluster around I see two pods that look healthy?

```
gpu-feature-discovery   nvgfd-node-feature-discovery-master-5ddf49b9cf-j2wlr   1/1     Running   0             108s
gpu-feature-discovery   nvgfd-node-feature-discovery-worker-4pfrc              1/1     Running   1 (74s ago)   108s
```

Reply:
Yep. I see them too. I thought need to run the upgrade for some reason. https://github.com/NVIDIA/gpu-feature-discovery?tab=readme-ov-file#installing-via-helm-install